store/sqlstore/post_store.go

```diff
@@ -1283,6 +1296,26 @@ func (s *SqlPostStore) search(teamId string, userId string, params *model.Search
 	return list, nil
 }
 
+func removeMysqlStopWordsFromTerms(terms string) (string, error) {
+	stopWords := make([]string, len(searchlayer.MYSQL_STOP_WORDS))
+	copy(stopWords, searchlayer.MYSQL_STOP_WORDS)
+	re, err := regexp.Compile(fmt.Sprintf(`^(%s)$`, strings.Join(stopWords, "|")))
```

Comment:
I've updated [the test](https://play.golang.com/p/NgkfNQkLFRN) and, if I'm not mistaken, the change is not working as expected

Reply:
yep, I had to change the approach because it couldn't find a regexp that could do all what I wanted, this is, a regexp that work with `"where about"` but not with `"where-about"` nor `"whereabout"`. I'm not the best with regexp but I thought I needed the lookbehind feature in golang regexp but it is not implemented, so I couldn't find a single regexp that solve these cases. After giving it some time, I decided to change the approch so now it is working good and it is covering all cases (at least those we talked about).

Reply:
Anyway the tests are passing now

https://github.com/rvillablanca/mattermost-server/blob/b42936f70941d44705b27f6b9f50bd8d152519ea/store/sqlstore/post_store_test.go#L22-L49

Reply:
Cool, looks good to me :)